src/m365/spo/commands/site/site-unarchive.spec.ts

```diff
@@ -0,0 +1,149 @@
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+
+    sinon.stub(cli, 'promptForConfirmation').callsFake(async () => {
+      promptIssued = true;
+      return false;
+    });
+    promptIssued = false;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      cli.promptForConfirmation,
+      request.post
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.connection.active = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.SITE_UNARCHIVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if url is not a valid SharePoint URL', async () => {
+    const actual = await command.validate({ options: { url: 'invalid' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation when a valid url is specified', async () => {
+    const actual = await command.validate({ options: { url: url } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('aborts unarchiving site when prompt not confirmed', async () => {
+    const postStub = sinon.stub(request, 'post');
+
+    await command.action(logger, { options: { url: url } });
+    assert(postStub.notCalled);
+  });
+
+  it('prompts before unarchiving the site when --force option is not passed', async () => {
+    await command.action(logger, { options: { url: url } });
+    assert(promptIssued);
+  });
+
+  it('unarchives the site when prompt confirmed', async () => {
+    sinonUtil.restore(cli.promptForConfirmation);
+    sinon.stub(cli, 'promptForConfirmation').resolves(true);
+
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `${adminUrl}/_vti_bin/client.svc/ProcessQuery`) {
+        return response;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { url: url, verbose: true } });
+    assert(postStub.calledOnce);
+  });
+
+  it('unarchives the site without prompting for confirmation when --force option specified', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `${adminUrl}/_vti_bin/client.svc/ProcessQuery`) {
+        return response;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { url: url, force: true, verbose: true } });
+    assert(postStub.called);
+  });
+
+  it('correctly handles error when unarchiving site that does not exist', async () => {
+    const errorMessage = 'File Not Found.';
+
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `${adminUrl}/_vti_bin/client.svc/ProcessQuery`) {
```

Comment:
When throwing an error, it's not really worth to check for the API URL. In this test we just want to test if an error is properly logged regardless of the URL that's passed.